src/components/ui-elements/Navbar.jsx

```diff
@@ -0,0 +1,57 @@
+import { useState, useEffect } from "react";
+import { Link } from "react-router-dom";
+import Button from "./Button";
+
+const Navbar = () => {
+  const [isDarkMode, setIsDarkMode] = useState(() => {
+    const savedMode = localStorage.getItem("isDarkMode");
+    return savedMode === "true" || false;
+  });
+
+  useEffect(() => {
+    if (isDarkMode) {
+      document.body.classList.add("dark-theme");
+    } else {
+      document.body.classList.remove("dark-theme");
+    }
+    localStorage.setItem("isDarkMode", isDarkMode);
```

Comment:
You can use `document.body.classList.toggle("dark-theme", isDarkMode);` by using the second argument of the `toggle` function


Also you should extract your logic of the localStorage into a service and expose methods from the service